pkg/app/piped/notifier/slack.go

```diff
@@ -110,9 +130,15 @@ func (s *slack) sendEvent(ctx context.Context, event model.NotificationEvent) {
 		}
 		return
 	}
-	if len(s.config.OAuthToken) != 0 {
+	if len(s.config.OAuthTokenData) != 0 {
 		if err := s.sendMessageViaAPI(ctx, msg); err != nil {
-			s.logger.Error(fmt.Sprintf("unable to send notification to slack: %v", err))
+			s.logger.Error(fmt.Sprintf("unable to send notification to slack using oauthTokenData: %v", err))
+		}
+		return
+	}
+	if len(s.config.OAuthTokenFile) != 0 {
+		if err := s.sendMessageViaAPI(ctx, msg); err != nil {
+			s.logger.Error(fmt.Sprintf("unable to send notification to slack using oauthTokenFile: %v", err))
```

Comment:
You should check if s.config.OAuthToken isn't nil

Reply:
added